src/js/utils/date-utils.js

```diff
@@ -0,0 +1,292 @@
+         *
+         * @return {string} either the entered or determined
+         * timezone, or (in the case of default) UTC
+         *
+         */
+        UserTimeZone = function(timezone) {
+            // linter wants a different variable
+            var returnzone;
+            if (timezone === undefined || timezone === 'None') {
+                returnzone = MomentTZ.tz.guess();
+            } else {
+                returnzone = timezone;
+            }
+            if (returnzone === undefined) {
+                return 'UTC';
+            }
+            return returnzone;
+        };
+
+        /**
+         *
+         * Attempt to transform time based on user pref/sys timezone
+         *
+         * @param {Object} dateobject, a moment.js datetime object
+         * @param {string} usertz, a timezone representation, e.g. 'America/New_York'
+         *
+         * @return {object} a moment.js datetime object in a determined timezone
+         */
+        LocalizedTime = function(dateobject, usertz) {
+            if (dateobject === undefined) {
+                return '';
+            }
+            if (usertz === undefined) {
+                return dateobject;
+            }
+            return dateobject.tz(usertz);
+        };
+
+        /**
+         * This generates the TZ-aware and language-pref displayed time string,
+         * with a set of built-ins for standard format across the platform
+         *
+         * the formatting is extensible, but this should suffice for most use cases
+         *
+         * e.g.
+         * preferred_language = 'en' yields Oct. 14, 2016 09:00 BST
+         * preferred_language = 'ru' yields 14 окт 2016 г. 09:00 BST
+         *
+         * Default to US-English
+         * (which is the current unlocalized view)
+         *
+         * @param {object} localizedtime a moment.js datetime object in a determined timezone
+         * @param {string} language: an ISO 639-1 language string (e.g. 'en', 'fr')
+         * @param {string} format: (optional) a desired format constant
+         *
+         * @return {string} a language/timezone aware localized string
+        */
+        TimeDisplay = function(localizedtime, language, format) {
+            /**
+             * This is a function to determine the language based on browser settings,
+             * but should probably be overridden by platform-set language preference
+             */
+            var displaylanguage;
+            var displayformat;
+            var displaytime;
+
+            if (language === undefined || language === 'None') {
+                displaylanguage = window.navigator.userLanguage || window.navigator.language;
+            } else {
+                displaylanguage = language;
+            }
+
+            /** default to en-us */
+            if (language === undefined) {
+                displaylanguage = 'en-US';
```

Comment:
Same type of comment here - `displaylanguage` is going through initialize as undefined->if some stuff assign a value->if still undefined use this default, you could go through initialize with default -> if some stuff assign a different value.

Also, consider using camelCase on your multi-word variable names for readability.